hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/ResponsePacketHandler.java

```diff
@@ -0,0 +1,15 @@
+package com.hazelcast.spi.impl.operationexecutor;
+
+import com.hazelcast.nio.Packet;
+import com.hazelcast.spi.impl.Response;
+
+/**
+ * The {@link com.hazelcast.spi.impl.operationexecutor.ResponsePacketHandler} is responsible for handling
+ * response packets.
+ */
+public interface ResponsePacketHandler {
+
+    Response deserialize(Packet packet) throws Exception;
```

Comment:
why is response handler responsible for de-serialization? I know there might be a good reason, but it's not clear from this contract.